### PR TITLE
Rdf fitting improvement

### PR DESCRIPTION
**Pull request by orionarcher**
## Description
This PR implements a new default solvation cutoff identifier based on `scipy.signal.find_peaks`. The performance seems to be better than the old method and it is vulnerable to fewer weird gotchas.

Based on @hmacdope's work in the `rdf_fitting_demo` tutorial.

Once merged, this PR will solve issue #43

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] implement new rdf parser based on find peaks
  - [x] add tests for parser
  - [x] add documentation for parser
  - [x] update tutorial

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by orionarcher** [comment]
Hi @hmacdope, I added a new way of finding the solvation cutoff based on `scipy.signal.find_peaks`.

    This function applies the following procedure to identify peaks.
        1. normalize the RDF
        2. apply a gaussian convolution (std=1.1) to the RDF
        3. call scipy.signal.find_peaks on the RDF and -1*RDF to find the peaks and troughs, respectively.
        4. apply simple heuristics to see if the cutoff is valid
              - trough follows peak
              - in `Solution.cutoff_region` (specified by kwarg)
              - normalized peak height > 0.05

Take a look when you get a chance, I think this is pretty much good to go!

File: solvation_analysis/tests/test_rdf_parser.py [hunk 1]
```diff
@@ -97,32 +94,74 @@ def test_good_cutoff(cutoff_region, cr_pts, cr_vals, expected):
...
+        equal_nan=True,
+    )
+
+
+def test_identify_cutoff_scipy_pf6(run_solution):
+    pf6_bins, pf6_data = run_solution.rdf_data["pf6"]
+    radius = identify_cutoff_scipy(pf6_bins, pf6_data, failure_behavior="warn"),
+    np.testing.assert_allclose(radius, 2.8, atol=0.2)
+    # fig, _ = plot_scipy_find_peaks(pf6_bins, pf6_data)
+    # fig.show()
```
Comment (hmacdope) [inline solvation_analysis/tests/test_rdf_parser.py:163]:
```suggestion

```

File: tutorials/rdf_fitting_demo.ipynb [hunk 2]
```diff
@@ -22,11 +22,11 @@
     "import numpy as np\n",
```
Comment (hmacdope) [inline tutorials/rdf_fitting_demo.ipynb:22]: <p>Do we want to keep this example for now but change it to something else down the line now that the <code>scipy</code> method is actually in the library?  I will raise an issue. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MDAnalysis/solvation-analysis/pull/63/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (orionarcher) [reply tutorials/rdf_fitting_demo.ipynb:22]: Though we now use scipy under the hood, it's still an excellent tutorial on implementing your own rdf kernel. 

The internal implementation is slightly different too, so I don't see a big need to change it.

**Comment by orionarcher** [comment]
Thanks for the comments! I implemented all of your suggested changes. Once this passes tests, I plan to merge.

**Pull request merged**